### PR TITLE
 Add cos7 for nodejs upgrade.

### PR DESCRIPTION
**Pull request by benjaminrwilson**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by benjaminrwilson** [comment]
@jakirkham and @wolfv -- this should be good to go. Thanks for the solution, @xhochy.

**Comment by jakirkham** [comment]
Hi Benjamin, thanks for the PR. Can you explain a bit more why this is needed? `nodejs` is only a build dependency for us. So shouldn't really affect users installing this package.

**Comment by benjaminrwilson** [comment]
Hi @jakirkham, this is to fix the failing build for the `osx_arm64` migration: https://github.com/conda-forge/binaryen-feedstock/pull/43 (this PR is a fork of the bot's branch).

**Comment by wolfv** [comment]
Looks good to me! @jakirkham waiting for you to give the go

**Comment by jakirkham** [comment]
Thanks for the info Benjamin!

In that case, we need to move this logic to `recipe/conda_build_config.yaml`. The `.ci_support/` directory is completely overwritten on each re-render (so the current changes would be lost). However re-rendering does pull from `recipe/conda_build_config.yaml` (better yet if we can move this to [`conda-forge-pinning`]( https://github.com/conda-forge/conda-forge-pinning-feedstock ))

**Comment by jakirkham** [comment]
Alternatively we can just move to CentOS 7 as @xhochy suggested here ( https://github.com/conda-forge/binaryen-feedstock/pull/43#issuecomment-932862858 )

**Comment by jakirkham** [comment]
Will need to re-render as well

**Comment by benjaminrwilson** [comment]
@conda-forge-admin please rerender

File: .ci_support/linux_64_nodejs16.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+channel_sources:
+- conda-forge
+channel_targets:
+- conda-forge main
+cxx_compiler:
+- gxx
+cxx_compiler_version:
+- '9'
+docker_image:
+- quay.io/condaforge/linux-anvil-cos7-cuda:10.2
```
Comment (jakirkham) [inline .ci_support/linux_64_nodejs16.yaml:16]: This seems odd. Not that it should cause any issues,  but wouldn't expect this image to be used here

Comment (benjaminrwilson) [reply .ci_support/linux_64_nodejs16.yaml:16]: Yeah, that's weird. The changes I initially pushed to this PR were from the rerender with `cos7` added (somehow I managed to push without actually modifying `conda-forge.yml`). AFAIK, this is an automatic change from `conda-smithy`.

Comment (xhochy) [reply .ci_support/linux_64_nodejs16.yaml:16]: This is a known issue with smithy that it selects the cuda image for cos7 :(

**Comment by xhochy** [comment]
This PR is LGTM from my side 👍

File: .ci_support/linux_64_nodejs16.yaml (hunk 1 above)
Comment (jakirkham) [reply .ci_support/linux_64_nodejs16.yaml:16]: Is there an existing issue tracking this somewhere @xhochy?

**Pull request merged**